### PR TITLE
[spark] Support spark to create external table sql statements.

### PR DESCRIPTION
**Pull request by zhuangchong**
### Purpose

Support spark to create external table sql statements.

```sql
CREATE EXTERNAL TABLE t1 (a INT, bb INT, c STRING)
```

### CODE REVIEW

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java [hunk 1]
```diff
@@ -659,12 +660,15 @@ public void createFormatTable(Identifier identifier, Schema schema) {
...
+    private boolean usingExternalTable(Map<String, String> tableOptions) {
         CatalogTableType tableType =
                 OptionsUtils.convertToEnum(
                         hiveConf.get(TABLE_TYPE.key(), CatalogTableType.MANAGED.toString()),
                         CatalogTableType.class);
-        return CatalogTableType.EXTERNAL.equals(tableType);
+
+        return CatalogTableType.EXTERNAL.equals(tableType)
+                || "TRUE"
+                        .equalsIgnoreCase(tableOptions.get(HIVE_EXTERNAL_TABLE_PROP.toLowerCase()));
```
Comment (JingsongLi) [inline paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java:671]: or maybe we should also get upper case?

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java [hunk 2]
```diff
@@ -135,6 +135,7 @@ public class HiveCatalog extends AbstractCatalog {
             "org.apache.paimon.hive.PaimonStorageHandler";
     private static final String HIVE_PREFIX = "hive.";
     public static final String HIVE_SITE_FILE = "hive-site.xml";
+    public static final String HIVE_EXTERNAL_TABLE_PROP = "EXTERNAL";
```
Comment (wwj6591812) [inline paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java:138]: private

Comment (zhuangchong) [reply paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java]: done.

File: paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java (hunk 1 above)
Comment (zhuangchong) [reply paimon-hive/paimon-hive-catalog/src/main/java/org/apache/paimon/hive/HiveCatalog.java]: done.

**Comment by wwj6591812** [comment]
+1

**Review by JingsongLi — APPROVED** [review]
+1

**Pull request merged**